package/MDAnalysis/analysis/base.py

```diff
@@ -447,8 +479,39 @@ def run(self, start=None, stop=None, step=None, frames=None,
             self.times[i] = ts.time
             self._single_frame()
         logger.info("Finishing up")
+        return self
+
+    def _setup_bslices(self, start=None, stop=None, step=None, frames=None):
+        self._bslices = [(start, stop, step, frames)]
+    
+    def _setup_scheduler(self, **kwa):
+        self._scheduler_kwargs = kwa
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, *, progressbar_kwargs={},
+            scheduler=None):
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(start=start, stop=stop, step=step, frames=frames, 
+            verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else: # elif scheduler is a type of dask scheduler
+            self._setup_scheduler()
```

Comment:
Raise an exception as soon as we enter a code path that requires dask but we don't have it imported.

Reply:
fixed